src/axom/inlet/tests/inlet_Inlet.cpp

```diff
@@ -1272,14 +1243,14 @@ TEST(inlet_verify, verifyFieldLambda)
   auto field2 = inlet->addString("field2");
   auto field3 = inlet->addString("NewTable/field3");
 
-  field2->registerVerifier([&]() -> bool {
-    auto str = (*inlet)["field2"].get<std::string>();
+  field2->registerVerifier([](Proxy& field) -> bool {
+    std::string str = field;
```

Comment:
```cpp
  field2->registerVerifier([](Proxy& field) -> bool {
    std::string str = field; 
    ...
  });
```
I get why it is implemented this way (i.e. passing in a Proxy to the lambda and having to then extract the actual type), but I'm wondering if this will be confusing for users...

I anticipate having to look up this syntax every time I need to use it.

Reply:
Also, should it be ``const Proxy& field`` since we don't want to change the value.
This is likely enforced by the Proxy type itself, but adding `const` might help the user understand the intent.

Reply:
I think for now it is sufficient to have the `Verifiable` signature use a `Table&` and `VerifiableScalar` use a `Field&` - initially I wanted the latter to inherit from the former but that would impose too much on the user.

I completely agree with the const correctness, but because no member functions in inlet are marked const, this would require a fair number of changes.  Const correctness is a primary goal for my next PR.